tests/suites/test_suite_nist_kw.function

```diff
@@ -7,151 +7,142 @@
  * END_DEPENDENCIES
  */
 
-/* BEGIN_CASE depends_on:MBEDTLS_SELF_TEST:MBEDTLS_AES_C */
-void mbedtls_nist_kw_self_test()
-{
-    TEST_ASSERT(mbedtls_nist_kw_self_test(1) == 0);
+/*
+ * Test vectors taken from NIST
+ * https://csrc.nist.gov/Projects/Cryptographic-Algorithm-Validation-Program/CAVP-TESTING-BLOCK-CIPHER-MODES#KW
+ */
+/* BEGIN_CASE depends_on:PSA_WANT_ALG_ECB_NO_PADDING */
+void mbedtls_nist_kw_self_test(int mode, data_t *kw_key,
+                               data_t *kw_msg, data_t *kw_res, int kw_out_len) {
+    unsigned char out[48];
+    size_t olen, out_len = kw_out_len;
+    mbedtls_svc_key_id_t key_id = MBEDTLS_SVC_KEY_ID_INIT;
+
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+
+    PSA_INIT();
+    psa_set_key_usage_flags(&attributes, PSA_KEY_USAGE_ENCRYPT |
+                        PSA_KEY_USAGE_DECRYPT);
+    psa_set_key_algorithm(&attributes, PSA_ALG_ECB_NO_PADDING);
+    psa_set_key_type(&attributes, PSA_KEY_TYPE_AES);
+    mbedtls_printf("\n");
+
+    PSA_ASSERT(psa_import_key(&attributes, kw_key->x, kw_key->len, &key_id));
+    PSA_ASSERT(mbedtls_nist_kw_wrap(key_id, mode, kw_msg->x,
+                               kw_msg->len, out, sizeof(out), &olen));
+    TEST_ASSERT(out_len == olen ||
+        memcmp(out, kw_res->x, out_len) == 0);
+    PSA_ASSERT(mbedtls_nist_kw_unwrap(key_id, mode,
+                                 out, olen, out, sizeof(out), &olen));
+
+    TEST_MEMORY_COMPARE(out, olen, kw_msg->x,  kw_msg->len);
+
+exit:
+    psa_destroy_key(key_id);
+    PSA_DONE();
 }
 /* END_CASE */
 
-/* BEGIN_CASE depends_on:MBEDTLS_AES_C */
+/* BEGIN_CASE depends_on:PSA_WANT_ALG_ECB_NO_PADDING */
```

Comment:
Same point about dependencies. (Multiple occurrences, I won't comment on each of them after this one.)